### DISCUSSION TITLE
Allow users to mount the preview application within any react application

### DISCUSSION THREAD

**Discussion by luisrudge** [resend/react-email#934] [Ideas]
### Goals

Today, Next.js is a hard dependency on the project. It requires everyone to install a full Next.js application in a subfolder (the `transactional` example). I'd like to be able to mount the emails in my own react application with a simple component.

### Background

I want to be able to use the same email preview app inside my own app

### Proposal

build the react application
```
yarn email build .....
```

```
import {EmailTemplates} from '.....'

createRoot(document.getElementById('app')).render(<EmailTemplates />);
  
```

**Comment by jinsley8** [comment]
1. Install `react-email` as a devDependency and you can add these scripts to `package.json`:
```
"scripts": {
        "email:dev": "email dev -p 4000",
        "email:export": "email export",
},
"devDependencies": {
        "react-email": "^1.9.5",
    },
```

2. Add `.react-email` to `.gitignore`

3. Add your email templates in an `/emails` folder

4. Now when you run `yarn email:dev` in development it will create the email previewer Next.js project in the `.react-email` folder but it won't be deployed to production and you can view the previewer at `http://localhost:4000`

**Comment by luisrudge** [comment]
@jinsley8 that's what I'm doing today (and the recommended approach by the react-email docs). What I'm saying is that I don't want to run another server. I want to render the component tree inside my app. I don't have enough context on why the team decided to use Next.JS for the preview, but this shouldn't really be a full fledge app, right? It should be a bunch of components and that's it. I just want to render the component tree in my app, instead of relying in a dependency that I'm not using it.

**Reply by jinsley8** [reply]
You can use the email templates without the preview app. Just don't install `react-email` or the scripts I mentioned.

All you need is:
```
"@react-email/components": "0.0.7",
"@react-email/render": "0.0.7",
```

**Reply by luisrudge** [reply]
Correct, and I'm doing this to render the previews for the design team: 
<img width="262" alt="image" src="https://github.com/resendlabs/react-email/assets/941075/10056a13-f6fc-498a-baf0-8883d805e2e1">

```tsx
  if (ENV !== 'production') {
    const EMAIL_TYPES = {
      PasswordChanged: PasswordChangedEmail,
      ResetPassword: ResetPasswordEmail,
    };
    app.get('/emails', async (req, res) => {
      const emails = Object.keys(EMAIL_TYPES)
        .map((key) => `<li><a href="/emails/${key}">${key}</a></li>`)
        .join('');
      return res.send(
        `<html>
          <body>
            <h1>Test Emails</h1>
            <ul>
              ${emails}
            </ul>
          </body>
          </html>`
      );
    });
    app.get('/emails/:template', async (req, res) => {
      const emails = Object.keys(EMAIL_TYPES);
      if (!emails.includes(req.params.template)) {
        return res.status(404).send('Not Found');
      }
      return res.send(render(EMAIL_TYPES[req.params.template]({} as any)));
    });
  }
```

My request was to be able to mount the preview application, not individual emails.

**Reply by ishworgiri1999** [reply]
The preview application is made with NextJS, but I'm not sure how you'd use it without NextJS. The only soln is to create your preview application and use render to create email outputs.

**Comment by gabrielmfern** [comment]
It's just as @ishworgiri1999 mentioned, there is no way we can render the preview server without using Next, or at least something that allows server actions and server components. The best you can go for is to render the emails yourself.
